### ISSUE TITLE
MahApps Integration Error with MahApps Style

### ISSUE THREAD

**Issue by RickStrahl** [ButchersBoy/Dragablz#100]
I'm trying to integrate Dragablz into a MapApps application and it looks like the functionality is perfect for what I'm looking for. However, I haven't had any luck getting Dragablz to respect MahApps theming which is key in order for me to use it.

Everything runs fine with the `Generic` theme, but when I add the `Mahapps` Theme in App.xaml I get: 

> Exeption thrown: 'System.Windows.Markup.XamlParseException' in PresentationFramework.dll
>
> Additional information: 'Initialization of 'Dragablz.TabablzControl' threw an exception.' Line number '14' and line position '30'.

if I load both:

```xml                
<ResourceDictionary Source="pack://application:,,,/Dragablz;component/Themes/Generic.xaml" />
<ResourceDictionary Source="pack://application:,,,/Dragablz;component/Themes/MahApps.xaml" />
```
it doesn't bomb but I still get no theme styling.

FWIW, I'm running the MapApps DarkTheme and I'm trying to get the tab pages to respect the theme colors (ie. dark background, light text). Dragablz is rendering everything in white background. 

I've also tried using `MetroTabItem` instead of plain `TabItem` but that also seems to have no effect. I realize I might be able to override the styling but the app already supports switching themes from MahApps so ideally I'd like to be able to just use that.

**Comment by ButchersBoy** [comment]
Hi Rick,

What you need to do is instruct dragablz to use the MahApps style, after you have referenced the MahApps resource dictionary.  The easiest way to do that is to use "BasedOn" in your App.xaml:

``` <Style TargetType="{x:Type dragablz:TabablzControl}" BasedOn="{StaticResource MahAppsTabablzControlStyle}" /> ```

There is a demo solution here: https://github.com/ButchersBoy/DragablzSamplez

Checkout the "StylezApp" project, more specifically, this line here:

https://github.com/ButchersBoy/DragablzSamplez/blob/master/StylezApp/App.xaml#L18

P.S.  great blog, I have referred to it many times in the past :smile:

**Comment by RickStrahl** [comment]
Ah, I missed that in the demo app. I think the sample I looked at didn't have this either because it was failing the same way as my app when I switched to the dark theme.

I ended up customizing the styles exactly the way I wanted them but that took me a while :-) WPF is not my forte. 

Here's what I ended up with which really nice but no longer themeable:
![tabs](https://cloud.githubusercontent.com/assets/1374013/15567500/8f196660-22db-11e6-8d1d-ef2a2eab130f.png)

which is a huge improvement of the stock tabs. But not themeable anymore :-). I'll take a look tomorrow and see what the default MahApps styling looks like.

Thanks for the kind words James - glad my blog has been useful. 

Appreciate the work with this control - pretty sure I'll be using it for other things as well in the future.
